Xamarin.Essentials/SecureStorage/SecureStorage.android.cs

```diff
@@ -259,20 +259,34 @@ internal byte[] Encrypt(string data)
             // Attempt to use GCMParameterSpec by default
             try
             {
-                cipher = Cipher.GetInstance(cipherTransformationSymmetric);
-                cipher.Init(CipherMode.EncryptMode, key, new GCMParameterSpec(128, iv));
+                try
+                {
+                    cipher = Cipher.GetInstance(cipherTransformationSymmetric);
+                    cipher.Init(CipherMode.EncryptMode, key, new GCMParameterSpec(128, iv));
+                }
+                catch (InvalidAlgorithmParameterException)
+                {
+                    // If we encounter this error, it's likely an old bouncycastle provider version
+                    // is being used which does not recognize GCMParameterSpec, but should work
+                    // with IvParameterSpec, however we only do this as a last effort since other
+                    // implementations will error if you use IvParameterSpec when GCMParameterSpec
+                    // is recognized and expected.
+                    cipher = Cipher.GetInstance(cipherTransformationSymmetric);
+                    cipher.Init(CipherMode.EncryptMode, key, new IvParameterSpec(iv));
+                }
             }
-            catch (InvalidAlgorithmParameterException)
+            catch (InvalidKeyException ike)
             {
-                // If we encounter this error, it's likely an old bouncycastle provider version
-                // is being used which does not recognize GCMParameterSpec, but should work
-                // with IvParameterSpec, however we only do this as a last effort since other
-                // implementations will error if you use IvParameterSpec when GCMParameterSpec
-                // is recognized and expected.
-                cipher = Cipher.GetInstance(cipherTransformationSymmetric);
-                cipher.Init(CipherMode.EncryptMode, key, new IvParameterSpec(iv));
+                if (retry)
+                {
+                    keyStore.DeleteEntry(alias);
```

Comment:
Does this mean that the current data will be lost whenever `InvalidKeyException` happens?